### PR TITLE
BUG: DCMTK reader wrongly rejects file with preamble

### LINKED ISSUE #4108: DCMTK reader fails for valid files
### Description

Valid dicom files are rejected.

### Steps to Reproduce

Use the itkDCMTKImageIO to read a file that has a dicom preamble and it is incorrectly rejected.

### Expected behavior

File should be accepted.

### Actual behavior

The `readNoPreambleDicom` implements a heuristic to see if the file starts with valid group/element tags.

https://github.com/InsightSoftwareConsortium/ITK/blame/master/Modules/IO/DCMTK/src/itkDCMTKImageIO.cxx#L231

But it fails when there is a preamble (which is often the case) there are 128 bytes and then the letters DICM.

https://dicom.nema.org/medical/dicom/current/output/html/part10.html#chapter_7

### Reproducibility

Always

### Versions

It appears to have been introduced by this commit:

https://github.com/InsightSoftwareConsortium/ITK/commit/19fa58feb02bfd45154e811d0e13510787e2adfb

### Environment

All platforms.

### Additional Information

I have a fix and will submit a PR.

### PR DESCRIPTION
**Pull request by pieper**
A previous optimization didn't take into account the premable of a dicom file.  This adds a check.

Fixes #4108

### CODE REVIEW

**Comment by pieper** [comment]
@hjmjohnson could you have a look at this?

It fixes https://github.com/InsightSoftwareConsortium/ITK/issues/4108

**Comment by issakomi** [comment]
I can confirm that DCMTKIO's _CanReadFile_ function returns **false** for valid DICOM files with preamble. I have reproduced it just now.

**Comment by jhlegarreta** [comment]
Thanks for doing this @pieper. I am willing to give this a :+1: when the following are addressed:
- We're trying to remove/have removed whitespaces before the scope resolution operator, e.g.
https://github.com/InsightSoftwareConsortium/ITK/pull/3664
so please remove them, e.g.
https://github.com/InsightSoftwareConsortium/ITK/pull/4110/files#diff-4f73091b3003136c994a445b6eda56d8ac3e9044936fbb78a0649164a8c31a73R68
- The output stream `<<` insertion operators were removed from `itkExceptionMacro` calls in: https://github.com/InsightSoftwareConsortium/ITK/pull/4057/files, so please remove them:
https://github.com/InsightSoftwareConsortium/ITK/pull/4110/files#diff-4f73091b3003136c994a445b6eda56d8ac3e9044936fbb78a0649164a8c31a73R94
If they should be added back, a separate commit and PR are required.
- Please document the method in the header file:
https://github.com/InsightSoftwareConsortium/ITK/pull/4110/files#diff-4f73091b3003136c994a445b6eda56d8ac3e9044936fbb78a0649164a8c31a73R135
- Adding a test that demonstrates the issue would be required to ensure that we do not run into this issue again.

In general, commits should only fall into one category, `BUG`, `DOC`, `STYLE`, etc. Helps reviewing, keeping changes of different nature separate, keeping a clean history, etc.

**Comment by pieper** [comment]
Thanks for the review @jhlegarreta - I had tried to add just the one function but it looks like my checkout was against an older target.  I'll fix it.

File: Modules/IO/DCMTK/src/itkDCMTKImageIO.cxx [hunk 1]
```diff
@@ -230,7 +259,7 @@ DCMTKImageIO::CanReadFile(const char * filename)
     }
```
Comment (thewtex) [inline Modules/IO/DCMTK/src/itkDCMTKImageIO.cxx:259]: This could be removed following seek to 0

**Review by thewtex — CHANGES_REQUESTED** [review]

**Comment by thewtex** [comment]
#4501 has updates with @jhlegarreta requests addressed (note that `readPreampleDicom` is not in the header, so it is moved to an anonymous namespace instead). Also, replaced the file re-open, close with a seek.

**Comment by thewtex** [comment]
Superceded by #4501

**Pull request closed without merging**